ee/desktop/user/server/server.go

```diff
@@ -152,6 +161,15 @@ func (s *UserServer) notificationHandler(w http.ResponseWriter, req *http.Reques
 	w.WriteHeader(http.StatusOK)
 }
 
+func (s *UserServer) showDesktop(w http.ResponseWriter, req *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+
+	sync.OnceFunc(func() {
+		s.showDesktopChan <- struct{}{}
+	})()
```

Comment:
Since this function is instantiated every time there's a request to /show, does `sync.OnceFunc()` enforce only sending to the channel once? Or we need to instantiate it earlier, like --

```
userServer := &UserServer{
    showFunc: sync.OnceFunc(func() {
		s.showDesktopChan <- struct{}{}
    }),
    ....
}

...

func (s *UserServer) showDesktop(w http.ResponseWriter, req *http.Request) {
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)

	s.showFunc()
}
```

Or maybe I'm misreading the `sync` documentation?

Reply:
yea, I think I'm doing it wrong =)